lib/validation.py

```diff
@@ -35,119 +35,391 @@ def _time_str(string):
+        # The time at which the run ended. This key will only exist if *status*
+        # is not equal to "in_progress".
+
+        "status": _status(),
+        # The state of this run, see the status schema below.
+
+        "aux": dict,
+        # A free-form dict that users can add custom information into. There are
+        # no constraints on the format of this dict, but it is recommended that
+        # users add their information to a sub-dict with a key that indicates
+        # its function. For example, to add information pertaining to a CI run,
+        # users might add a key called "continuous_integration_data" whose value
+        # is a sub-dict containing all required fields.
+
         "parallelism": voluptuous.Any({
+        # This dict contains information about the parallelism level of the jobs
+        # that litani runs. This is to measure whether the run is using as many
+        # processor cores as possible over the duration of the run.
+
             voluptuous.Optional("trace"): [{
+            # A list of samples of the run's concurrency level.
+
                 "time": _ms_time_str,
+                # The time at which the sample was taken.
+
                 "finished": int,
+                # How many jobs have finished
+
                 "running": int,
+                # How many jobs are running
+
                 "total": int,
+                # The total number of jobs
+
             }],
+
             voluptuous.Optional("max_parallelism"): int,
+            # The maximum parallelism attained over the run
+
             voluptuous.Optional("n_proc"): voluptuous.Any(None, int),
+            # The number of processors detected on this machine
+
         }),
+
         "pipelines": [{
+        # Each pipeline contains ci_stages which contain jobs.
+
             "url": str,
+
             "name": str,
-            "status": voluptuous.Any("in_progress", "fail", "success"),
+            # The pipeline name. The set of pipeline names are all the names
+            # passed to the --pipeline-name flag of *litani-add-job(1)*.
+
+            "status": _status(),
+            # The pipeline's state, see the status schema below.
+
             "ci_stages": [{
+            # Each ci_stage contains a list of jobs.
+
                 "url": str,
                 "complete": bool,
+                # Whether all the jobs in this stage are complete.
+
                 "name": voluptuous.Any("build", "test", "report"),
-                "status": voluptuous.Any("fail", "fail_ignored", "success"),
+                # The stage's name. In the future, it may be possible for users
+                # to pass arbitrary names here, so it is advisable not to assume
+                # that these are the only three permissable values.
+
+                "status": _outcome(),
+                # The stage's state, see the outcome schema below.
+
                 "progress": voluptuous.All(int, voluptuous.Range(min=0, max=100)),
+
```

Comment:
Pretty obvious what `progress` represents.

However, should a comment be added to clarify how a value for `progress` is determined by Litani?